borbskeeper/Features/ComposeTask/ComposeTaskViewController.m

```diff
@@ -75,27 +76,29 @@ - (IBAction)didTapSaveTask:(id)sender {
                                  withDueDate:self.taskDeadlineDatePicker.date];
             [BorbParseManager saveTask:newTask withCompletion:^(BOOL succeeded, NSError * _Nullable error) {
                 if (succeeded) {
+                    NSString *objectId = [newTask objectId];
+                    [PushNotifications createNotificationForTask:newTask WithID:objectId];
                     [self.delegate didSaveTask];
                     [self.navigationController dismissViewControllerAnimated:YES completion:nil];
                 } else {
                     [self presentViewController:saveNotSuccessfulAlert animated:YES completion:nil];
                 }
             }];
         } else {
-            //    have to change the already made properties in the task
             PFQuery *query = [PFQuery queryWithClassName:@"Task"];
             
-            NSString *objectId = self.task.objectId;
-            //    tried this but there is null in the objectId
+            NSString *objectId = [self.task objectId];
             [query getObjectInBackgroundWithId:objectId
                                          block:^(PFObject *task, NSError *error) {
                                              task[@"taskName"] = self.taskTitleTextField.text;
                                              task[@"taskDescription"] = self.taskDescTextView.text;
                                              task[@"dueDate"] = self.taskDeadlineDatePicker.date;
                                              [task saveInBackground];
+                                             
+                                             [PushNotifications createNotificationForTask:(Task *)task WithID:objectId];
                                          }];
-            
-            
+            [PushNotifications deleteNotificationForTaskWithID:objectId];
```

Comment:
why do we delete the notification here?

Reply:
When the user updates a task, the notification associated with that task must be changed too. However, there is no way to update an existing notification; it must be deleted and a new one must be created. 
The code deletes the notification associated with the before-update task, and in the completion block above a new notification is created for the after-update task.